### PR TITLE
Implemented private remote tests

### PR DESCRIPTION
**Pull request by implausible**
Added test to check if nodegit can fetch from a private repo with valid creds

Added test to check if nodegit cannot fetch from private repo with invalid creds

### CODE REVIEW

**Comment by tbranyen** [comment]
Wow thanks!

File: test/tests/remote.js [hunk 1]
```diff
@@ -163,6 +164,63 @@ describe("Remote", function() {
...
+      }
+    };
+
+    return remote.fetch(null, fetchOptions, "Fetch from private")
+      .catch(function() {
+        assert.fail("Unable to fetch from private repository");
+      });
+  });
+
+  it("can reject fetching from private repository without valid " +
```
Comment (tbranyen) [inline test/tests/remote.js]: One minor thing here.  Is it necessary to concat here?

Comment (implausible) [inline test/tests/remote.js]: Changed so it that function () { is on the next line, it's necessary to split up the it("string" and the function definition. What's preferable? it("", \nfunction\n{ or it("", \nfunction() { ?

Comment (tbranyen) [inline test/tests/remote.js]: If you keep it on the same line does it fail JSHint or something?

Comment (implausible) [inline test/tests/remote.js]: yup

Comment (tbranyen) [inline test/tests/remote.js]: Okay, oh well ha.  Thanks!

**Pull request merged**